src/simcore_service_deployment_agent/git_url_watcher.py

```diff
@@ -138,7 +157,9 @@ async def _update_repository(repo: GitRepo):
 async def _init_repositories(repos: List[GitRepo]) -> Dict:
     description = {}
     for repo in repos:
-        repo.directory = tempfile.TemporaryDirectory().name
+        with tempfile.TemporaryDirectory() as tmpfile:
+            directoryName = copy.deepcopy(tmpfile)
+        repo.directory = directoryName
```

Comment:
if I get this correctly, you create a temporary directory which you receive as a Path in tmpfile. then you deepcopy that filepath. the directory gets deleted when going out of the context manager. and you keep the name out.
Would it not be simpler to use ```tempfile.mkdtemp``` ? you could also use ```tempfile.gettempdir()``` to get to the /tmp folder and then create a folder there?
Also be careful that the context manager might not trigger the deletion of the folder right away, but only when the garbage collector runs.

Reply:
you are right, I refactored using `tempfile.mkdtemp`, thx